src/session_impl.cpp

```diff
@@ -5619,9 +5621,16 @@ namespace aux {
 
 	void session_impl::stop_dht()
 	{
-		if (!m_dht) return;
-		m_dht->stop();
-		m_dht.reset();
+		if (m_dht)
+		{
+			m_dht->stop();
+			m_dht.reset();
+		}
+
+		if (m_dht_storage)
```

Comment:
I don't think you should check if m_dht_storage is set. just call reset() unconditionally